odabot/cli.py

```diff
@@ -651,7 +661,267 @@ def verify_workflows(obj):
+                               res.status_code, 
+                               res.text)
         
+        res = requests.post(api_url, json=data, headers=headers)
+        
+        if res.status_code != 201:
+            raise RuntimeError('Error creating PR. Status: %s. Response text: %s', 
+                               res.status_code, 
+                               res.text)
+        else:
+            logger.info(f"New PR {res.json()['html_url']}")
+            return res.json()
+        
+    
+    
+    
+        
+    if "deployed_tools" not in oda_bot_runtime:
+        oda_bot_runtime["deployed_tools"] = {}
+    deployed_tools = oda_bot_runtime["deployed_tools"]
+
+    git_clone_or_update(tools_repo_dir, tools_repo, target_branch)
+    os.chdir(tools_repo_dir)
+    sp.run(['git', 'config', 'user.name', git_name], check=True)
+    sp.run(['git', 'config', 'user.email', git_email], check=True)
+    sp.run(['git', 'config', 'credential.helper', f'store --file={git_credentials}'])
+    
+    while True:
+        git_clone_or_update(tools_repo_dir, tools_repo, target_branch)
+        try:
+            for project in requests.get(f'{renkuapi}groups/{renku_gid}/projects?include_subgroups=yes&order_by=last_activity_at').json():
+                try:    
+                    if re.match(pattern, project['name']) and 'galaxy-tool' in project['topics']:
+                        logger.info("%20s %s", project['name'], project['http_url_to_repo'])
+                        logger.debug("%s", json.dumps(project))
+
+                        last_commit = requests.get(f'{renkuapi}projects/{project["id"]}/repository/commits?per_page=1&page=1').json()[0]
+                        last_commit_created_at = last_commit['created_at']
+
+                        logger.info('last_commit %s from %s', last_commit, last_commit_created_at)
+                        
+                        saved_last_commit_created_at = deployed_tools.get(project['http_url_to_repo'], {}).get('last_commit_created_at', 0)
+                        #saved_last_tool_version = deployed_tools.get(project['http_url_to_repo'], {}).get('last_tool_version', '0.0.0+galaxy0')
+                        
+                        logger.info('last_commit_created_at %s saved_last_commit_created_at %s', last_commit_created_at, saved_last_commit_created_at )
+
+                        if last_commit_created_at == saved_last_commit_created_at and not force:
+                            logger.info("no need to deploy this tool")
+                        else:
+                            wf_repo_dir = os.path.join(repo_cache_dir, project['path'])
+                            git_clone_or_update(wf_repo_dir, project['http_url_to_repo'])
+                        
+                            req_file = os.path.join(wf_repo_dir, 'requirements.txt') if os.path.isfile(os.path.join(wf_repo_dir, 'requirements.txt')) else None
```

Comment:
this is a very long line. And the indent is very deep too. Could it be taken into a function?